File: TypeScript/8-kyu/Total amount of points.ts
export function points(games : string[]): number {
  return games.reduce((total, game) => {
    const [x, y] = game.split(':').map(Number);
    
    if (x > y) {
      return total += 3;
    } else if (x === y) {
      return total += 1;
    } else {
      return total;
    }
  }, 0);
}

File: Go/8-kyu/Total amount of points.go
package kata

import (
  "strconv"
  "strings"
)

func Points(games []string) int {
    points := 0

    for _, game := range games {
        score := strings.Split(game, ":")
        x, _ := strconv.Atoi(score[0])
        y, _ := strconv.Atoi(score[1])

        if x > y {
            points += 3 
        } else if x < y {
            points += 0 
        } else {
            points += 1 
        }
    }

    return points
}
